runtime/jcl/common/sun_reflect_ConstantPool.c

```diff
@@ -564,35 +564,62 @@ Java_sun_reflect_ConstantPool_getMemberRefInfoAt0(JNIEnv *env, jobject unusedObj
 				result = getROMCPItem(vmThread, constantPoolOop, classRefCPIndex, J9CPTYPE_CLASS, (J9ROMConstantPoolItem **) &classRef);
 				if (OK == result) {
 					J9UTF8 *className = J9ROMCLASSREF_NAME(classRef);
+					j9object_t internalNameObject = NULL;
+					j9object_t internalSignatureObject = NULL;
 					j9object_t internalClassNameObject = gcFunctions->j9gc_createJavaLangString(vmThread, J9UTF8_DATA(className), (U_32) J9UTF8_LENGTH(className), 0);
-					j9object_t internalNameObject = gcFunctions->j9gc_createJavaLangString(vmThread, J9UTF8_DATA(name), (U_32) J9UTF8_LENGTH(name), 0);
-					j9object_t internalSignatureObject = gcFunctions->j9gc_createJavaLangString(vmThread, J9UTF8_DATA(signature), (U_32) J9UTF8_LENGTH(signature), 0);
+					if (internalClassNameObject == NULL) {
+						goto dropAccess;
+					}
 					classNameObject = vmFunctions->j9jni_createLocalRef(env, internalClassNameObject);
+					if (classNameObject == NULL) {
+						goto dropAccess;
+					}
+					internalNameObject = gcFunctions->j9gc_createJavaLangString(vmThread, J9UTF8_DATA(name), (U_32) J9UTF8_LENGTH(name), 0);
+					if (internalNameObject == NULL) {
+						goto dropAccess;
+					}
 					nameObject = vmFunctions->j9jni_createLocalRef(env, internalNameObject);
+					if (nameObject == NULL) {
+						goto dropAccess;
+					}
+					internalSignatureObject = gcFunctions->j9gc_createJavaLangString(vmThread, J9UTF8_DATA(signature), (U_32) J9UTF8_LENGTH(signature), 0);
+					if (internalSignatureObject == NULL) {
+						goto dropAccess;
+					}
 					signatureObject = vmFunctions->j9jni_createLocalRef(env, internalSignatureObject);
+					if (signatureObject == NULL) {
+						goto dropAccess;
+					}
 				}
 			}
 		}
+dropAccess:
 		vmFunctions->internalExitVMToJNI(vmThread);
 	}
 
 	if ((NULL != classNameObject) && (NULL != nameObject) && (NULL != signatureObject)) {
 		jclass jlString = JCL_CACHE_GET(env, CLS_java_lang_String);
 		jobject array = (*env)->NewObjectArray(env, 3, jlString, NULL);
+		if (array == NULL) {
+			goto done;
+		}
 		(*env)->SetObjectArrayElement(env, array, 0, classNameObject);
+		if ((*env)->ExceptionCheck(env)) {
```

Comment:
This is a waste - the previous call cannot fail.

Reply:
Theory was that jni check would warn on this code without the exception checks

Reply:
I don't think it will (I suspect we have lots of other places that are not doing the check).

Reply:
Turns out that jni check will warning for a pending exception when calling `SetObjectArrayElement` but it doesn't set the `jniCheckSetPotentialPendingException` flag

Reply:
My mistake - it can throw exceptions (though you have to make a programming error to trigger it).